FaunaDB.Client/Utils/CheckLatestVersion.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FaunaDB.Client.Utils
+{
+    public class CheckLatestVersion
+    {
+        private const string packageName = "FaunaDB.Client";
+        public static bool? AlreadyChecked { get; set; } = null;
+        public static async Task GetVersionAsync()
+        {
+            if (AlreadyChecked.HasValue) return;
+            var latestNuGetVesrionString = string.Empty;
+            var url = $"https://api.nuget.org/v3-flatcontainer/{packageName}/index.json";
+            try
+            {
+                var httpClient = new HttpClient();
+#if NET45
+                ServicePointManager.SecurityProtocol = SecurityProtocolType.Ssl3 | SecurityProtocolType.Tls | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls12;
+#endif
+                var response = await httpClient.GetAsync(url);
+                string versionsResponse = await response.Content.ReadAsStringAsync();
+                Newtonsoft.Json.Linq.JObject jObject = Newtonsoft.Json.Linq.JObject.Parse(versionsResponse);
```

Comment:
this is just a question:
why do we need to use the full name of the class, instead of placing it above to `using`?

Reply:
Just didn't wont to use link to Newtonsoft. I remember we want to move to Text.Json.